build.gradle

```diff
@@ -240,38 +240,46 @@ tasks.distTar {
 
 tasks.register('validateDependencies') {
     doLast {
-        def dependencyVersions = [:]
         def conflictsFound = false
 
-        subprojects { subproject ->
-            println "Module: ${subproject.name}"
-            def runtimeClasspath = subproject.sourceSets.main.runtimeClasspath
-            runtimeClasspath.each { file ->
-
-                def fileName = file.name
-                def matcher = fileName =~ /(.+)-(\d+\.\d+\.\d+(-.+)?)(\..+)/
-
-                if (matcher) {
-                    def artifactId = matcher[0][1]
-                    def version = matcher[0][2]
-
-                    if (dependencyVersions.containsKey(artifactId)) {
-                        if (dependencyVersions[artifactId] != version) {
-                            println "Conflict found for $artifactId: ${dependencyVersions[artifactId]} vs $version"
-                            conflictsFound = true
+        def storages = ["s3", "gcs", "azure"]
+        def ignoreDeps = ["e2e"]
+
+        storages.forEach { s ->
+            println "Validating modules for storage: $s"
+            def dependencyVersions = [:]
+            subprojects { subproject ->
+                if (ignoreDeps.contains(subproject.name)) return
+                if (s == subproject.name || !storages.contains(subproject.name)) {
```

Comment:
How do you thinks: would it be possible to declare this task on the storage project-level so we don't need to filter and search (of course from the single definition)?

Reply:
Good idea, there's only common and core modules to add as deps to storage backends. See last commit, thanks!